iceoryx_dds/include/iceoryx_dds/dds/data_writer.hpp

```diff
@@ -27,44 +27,36 @@ namespace dds
 {
 ///
 /// @brief Abstraction for DDS Data Writers.
-/// Provides the minimum functionality required for posh-dds gateway implementations.
+/// @note Provides the minimum functionality required for posh-dds gateway implementations.
 ///
 class DataWriter
 {
   public:
+    /// @brief Virtual default destructor
```

Comment:
@dkroenke do we really have to document this. Comments like
`@brief default construct` or `@brief default destructor` deliver absolutely no value because it's exactly what the signature says since it is the default constructor or the default destructor. Here again, I would use common sense. The comment should be helpful for the developer and not clutter the code which can be counterproductive.

Otherwise, we end up with stuff like this:
```cpp
// assign zero to integer a
int a = 0;
```